Need extra = in assert in d4swap.c

Just noticed this (I think new) warning:

```
d4swap.c:238:12: warning: suggest parentheses around assignment used as truth value [-Wparentheses]
     assert(basetype->sort = NCD4_TYPE);

```
Probably meant to have another = there. @DennisHeimbigner is this in some of the code you just had merged?

I have a PR up now. I cleared a couple more warnings as well in the same directory.

Here's some I don't know what to do about:

```
d4meta.c: In function ‘compileAttrValues’:
d4meta.c:732:16: warning: pointer targets in assignment differ in signedness [-Wpointer-sign]
     if((memory = (char*)d4alloc(count*size))==NULL)
                ^
d4meta.c: In function ‘convertString’:
d4meta.c:783:18: warning: format ‘%lld’ expects argument of type ‘long long int *’, but argument 3 has type ‘long long int (*)[1]’ [-Wformat=]
  if(sscanf(s,"%lld",&converter->i64) != 1) return THROW(NC_ERANGE);
               ~~~^  ~~~~~~~~~~~~~~~
d4meta.c:789:18: warning: format ‘%llu’ expects argument of type ‘long long unsigned int *’, but argument 3 has type ‘long long unsigned int (*)[1]’ [-Wformat=]
  if(sscanf(s,"%llu",&converter->u64) != 1) return THROW(NC_ERANGE);
               ~~~^  ~~~~~~~~~~~~~~~
d4meta.c:793:17: warning: format ‘%lf’ expects argument of type ‘double *’, but argument 3 has type ‘double (*)[1]’ [-Wformat=]
  if(sscanf(s,"%lf",&converter->f64) != 1) return THROW(NC_ERANGE);

```               ~~^  ~~~~~~~~~~~~~~~
So I didn't touch this one.

Dennis, do you think I can just cast away the warnings in the printf? I believe that casting will remove the warning without changing code behavior, correct?

--- a/libdap4/d4chunk.c
+++ b/libdap4/d4chunk.c
@@ -65,7 +65,7 @@ NCD4_dechunk(NCD4meta* metadata)
 	/* setup as dmr only */
 	metadata->serial.dmr = (char*)metadata->serial.rawdata; /* temp */
 	metadata->serial.dmr[metadata->serial.rawsize-1] = '\0';
-	metadata->serial.dmr = strdup(q);
+	metadata->serial.dmr = strdup((char *)q);
 	if(metadata->serial.dmr == NULL)
 	    return THROW(NC_ENOMEM);
 	return THROW(NC_NOERR); 

--- a/libdap4/d4data.c
+++ b/libdap4/d4data.c
@@ -180,7 +180,6 @@ fillstruct(NCD4meta* meta, NCD4node* type, void** offsetp, void** dstp, NClist* 
  
 #ifdef CLEARSTRUCT
     /* Avoid random data within aligned structs */
-    d4size_t prevoffset = 0;    
     memset(dst,0,type->meta.memsize);
 #endif
 

--- a/libdap4/d4file.c
+++ b/libdap4/d4file.c
@@ -442,7 +442,6 @@ fail:
 static void
 applyclientparamcontrols(NCD4INFO* info)
 {
-    NCD4meta* meta = info->substrate.metadata;
     const char* value;
 
     /* clear the flags */

--- a/libdap4/d4rc.c
+++ b/libdap4/d4rc.c
@@ -86,7 +86,7 @@ rctrim(char* text)
 static NClist*
 rcorder(NClist* rc)
 {
-    int i,j;
+    int i;
     int len = nclistlength(rc);
     NClist* newrc = nclistnew();
     if(rc == NULL || len == 0) return newrc;

--- a/libdap4/d4swap.c
+++ b/libdap4/d4swap.c
@@ -235,7 +235,7 @@ walkSeq(NCD4meta* compiler, NCD4node* topvar, NCD4node* vlentype, void** offsetp
         swapinline64(&recordcount);
 
     basetype = vlentype->basetype; /* This may be of any type potentially */
-    assert(basetype->sort = NCD4_TYPE);
+    assert(basetype->sort == NCD4_TYPE);
 
     for(i=0;i<recordcount;i++) {
         switch(basetype->subsort) {

